static/components/List.js

```diff
@@ -1,78 +1,85 @@
 import React, { Component } from 'react';
-import ListItem from './listitem';
-import uuidv4 from 'uuid/v4';
+import ListItem from './ListItem';
+import uuidv4 from 'uuid/v4'; // generate unique id
 
 class List extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            items: []
+            items: [],
         };
-        this.removeItem = this.removeItem.bind(this)
+        this.removeItem = this.removeItem.bind(this);
+        this.editItem = this.editItem.bind(this);
     }
 
-    onSubmit = (event) => {
-        event.preventDefault();
-        const newValue = this.input.value;
-        
-        if(newValue == '') {
-            return false;
-        } else {
-            var newItem = {
-                text: newValue,
-                date: new Date().toLocaleTimeString(),
-                id: uuidv4()
-            }
-            var newItems = [...this.state.items, newItem];
+    addTodo = (props) => {
```

Comment:
This is an example of a bad design. This method copes with both adding new todo and updating existing, which rudely violates single responsibility principle. Please read about that and refactor this for separate methods

Reply:
Moreover, the method is called addTodo, but it actually not only adds todo, but also updates. It can be unexpected surprise to your colleagues)